util/topgen/intermodule.py

```diff
@@ -79,6 +73,21 @@ def elab_intermodule(topcfg: OrderedDict):
     # intermodule
     definitions = []
 
+    # Check the originator
+    # As inter-module connection allow only 1:1, 1:N, or N:1, pick the most
+    # common signals. If a requester connects to multiple responders/receivers,
+    # the requester is main connector so that the definition becomes array.
+    #
+    # For example:
+    #  inter_module: {
+    #    'pwr_mgr.pwrup': ['lc.pwrup', 'otp.pwrup']
+    #  }
+    # The tool adds `struct [1:0] pwr_mgr_pwrup`
+    # It doesn't matter whether `pwr_mgr.pwrup` is requester or responder.
+    # If the key is responder type, then the connection is made in reverse,
+    # such that `lc.pwrup --> pwr_mgr.pwrup[0]` and
+    # `otp.pwrup --> pwr_mgr.pwrup[1]`
+
```

Comment:
it feels a little like the `req` / `rsp` names here are overloaded...
Am i understanding correctly that when you say `req` in this context, you're really talking about the key in the intermodule definition, and rsp is the value?

because the comment above also says it doesn't matter if it's requester or responder...so i feel like the terms are getting a bit jumbled.  Am I understanding this right? It almost feels like we should use a new term for the stuff below..like "src/dest" or something.

Reply:
src/dest seems like a poor name too...since it implies someone is always driving the net and someone is always receiving..

if i understand what you're saying, the "key" is really just the holder of the array object. It may be 'requestor' (in which case it would drive the array) or it may be 'responder' (in which case its array values would be driven).  Is this roughly the right understanding?